common/scala/src/main/scala/org/apache/openwhisk/core/entity/ActivationResult.scala

```diff
@@ -62,40 +63,47 @@ protected[core] object ActivationResponse extends DefaultJsonProtocol {
   val ApplicationError = 1 // action ran but there was an error and it was handled
   val DeveloperError = 2 // action ran but failed to handle an error, or action did not run and failed to initialize
   val WhiskError = 3 // internal system error
+  val Truncated = 4 // action ran correctly, but returned a response that was too large
```

Comment:
I think we can avoid this by treating truncation as ApplicationError

Reply:
I thought knowing the difference is useful as there might be other areas of code that might want to know that the response was really big.

Reply:
I think the result when truncated carries a message to that effect. I'd rather not add another code as well if we can avoid it.

Reply:
Okay, seems like we're losing information in the error handling that could be useful.

For example the activation storage may want to not persist huge responses but application errors maybe more interesting. To do this they'll need to now parse the error string which doesn't any exported contract, but perhaps there is another path for detecting that which I don't know about.

I'll back it out the new code.

Reply:
That's true and a good point. How are you anticipating making a distinction between large input that should persisted and one that shouldn't?

Reply:
Okay, seems like we're losing information in the error handling that could be useful.

For example the activation storage may want to not persist huge responses but application errors maybe more interesting. To do this they'll need to now parse the error string which doesn't any exported contract, but perhaps there is another path for detecting that which I don't know about.

I'll back it out the new code.

Reply:
@rabbah I don't know the answer on that one --- I just remember hearing that someone was planning to try put less in the DB and this seemed like an obvious candidate.

Reply:
K. I'm OK with this change thinking about it some more. Perhaps solicit some input from the dev list before backing it out?

Reply:
Will do, thanks!

Reply:
Currently in https://github.com/apache/openwhisk/blob/ce6a5b7611c46f7d8a528c9987cfb9e9d50ade44/common/scala/src/main/scala/org/apache/openwhisk/core/entity/ActivationResult.scala#L208 there is precisely one path that results in ApplicationError (where developer returns an `{error:"message"}` in result), and I am suggesting we change this method to include one additional case here, where truncation results in ApplicationError as well, with the difference that we force the error message to be the truncation message. The result is that all other code remains the same, but the container is left in place to continue processing. You might wonder when the result of returning an error message in result AND truncating the result - but we also don't handle that case today (truncation error takes precedent). 

There was a separate problem with truncation, in that we currently truncate to the "max size string", which is mostly useless afaik, since this is often (always?) json or some other content that cannot be parsed at all. I wouldn't mind seeing the truncation amount limited to either 0 or 512 bytes (or something small) so that the extra useless content is not contributing to db bloat - since the truncated content is arguably useless, even for debugging purposes.

Reply:
This comes down to is it useful to know that an action ended with a truncation in other parts of the code. If not then sure I can swap to have truncation raise an ApplicationError instead of DeveloperError, if it would be helpful to understand the different than we should leave in the new Truncate Error.

The truncate error being not helpful is a second question we can easily change the implementation of `truncatedResponse`.

Reply:
imo it's not necessary for other parts of the code to know, at least not in step 1. Feels like we should make this an application error first and maybe discuss on the dev list if people feel that there's a need to carry it being a TruncationError further into the codebase. That could then quite simply be a second PR?

Reply:
@markusthoemmes already started a thread for that on the DL. So, with 3 against although I convinced Rodric a bit I'll just back out the new code.